p2p/src/network/pubsub/p2p_network_pubsub_actions.rs

```diff
@@ -91,32 +101,65 @@ pub enum P2pNetworkPubsubAction {
 
     /// An error occured during the signing process.
     #[action_event(level = warn, fields(display(author), display(topic)))]
-    SignError { author: PeerId, topic: String },
+    SignError {
+        author: PeerId,
+        topic: String,
+    },
 
     /// Finalize the broadcasting of a signed message by attaching the signature.
     ///
     /// **Fields:**
     /// - `signature`: The cryptographic signature of the message.
-    BroadcastSigned { signature: Data },
+    BroadcastSigned {
+        signature: Data,
+    },
 
     /// Prepare an outgoing message to send to a specific peer.
-    OutgoingMessage { peer_id: PeerId },
+    OutgoingMessage {
+        peer_id: PeerId,
+    },
 
     /// Clear the outgoing message state for a specific peer after sending.
-    OutgoingMessageClear { peer_id: PeerId },
+    OutgoingMessageClear {
+        peer_id: PeerId,
+    },
 
     /// An error occured during the sending of an outgoing message.
     ///
     /// **Fields:**
     /// - `msg`: The protobuf message that failed to send.
     #[action_event(level = warn, fields(display(peer_id), debug(msg)))]
-    OutgoingMessageError { msg: pb::Rpc, peer_id: PeerId },
+    OutgoingMessageError {
+        msg: pb::Rpc,
+        peer_id: PeerId,
+    },
 
     /// Send encoded data over an outgoing stream to a specific peer.
     ///
     /// **Fields:**
     /// - `data`: The encoded data to be sent.
-    OutgoingData { data: Data, peer_id: PeerId },
+    OutgoingData {
+        data: Data,
+        peer_id: PeerId,
+    },
+
+    BroadcastValidationCallback {
+        message_id: P2pNetworkPubsubMessageCacheId,
+    },
+
+    BroadcastValidatedMessage {
+        message_id: BroadcastMessageId,
+    },
+    HandleIncomingMessage {
+        message: pb::Message,
+        message_content: GossipNetMessageV2,
+        peer_id: PeerId,
+    },
+    /// Delete expired messages from state
+    PruneMessages {},
+    RejectMessage {
+        message_id: BroadcastMessageId,
```

Comment:
I would add a "reason" here (can be a static string). Consider that this will show up in logs and it is useful to know why it happened (for debugging purposes). Same if we are writing tests for specific cases and want to ensure that it was rejected for the right reason.

Same for the `IgnoreMessage` action I mentioned in another comment, the fact that it was ignored must be reflected in an action (even if it does nothing), otherwise it is invisible (both for test-writting and debugging purposes).

Reply:
> even if it does nothing
that will probably not be the case btw, we can use the `Ignore` action to discard the contents and keep just the message id around so that we can recognize that we have seen it already if we get that same message again.